floris/tools/optimization/layout_optimization/layout_optimization_base.py

```diff
@@ -30,12 +35,19 @@ def __init__(self, fi, boundaries, min_dist=None, freq=None, enable_geometric_ya
         else:
             self.min_dist = min_dist
 
-        # If freq is not provided, give equal weight to all wind conditions
-        if freq is None:
-            self.freq = np.ones((self.fi.floris.flow_field.n_findex,))
-            self.freq = self.freq / self.freq.sum()
+        # If wind_data is None, make a simple time series object off the fi
+        if wind_data is None:
+            wind_data = TimeSeries(wind_directions=fi.wind_directions,
+                                wind_speeds=fi.wind_speeds,
+                                turbulence_intensities=fi.turbulence_intensities)
         else:
-            self.freq = freq
+            if ((not isinstance(wind_data, WindDataBase)) and
+                 (not isinstance(wind_data, TimeSeries)) and
+                 (not isinstance(wind_data, WindRose))):
```

Comment:
I think we only need to check that wind_data is an instance of WindDataBase

Reply:
Fixed!